pkg/reconciler/v1alpha1/route/traffic/traffic.go

```diff
@@ -87,13 +93,31 @@ func BuildTrafficConfiguration(configLister listers.ConfigurationLister, revList
 	return builder.build()
 }
 
+// SubrouteDomain returns the domain name of a traffic target given the traffic target name and the Route's base domain.
+func SubrouteDomain(name, domain string) string {
+	if name == DefaultTarget {
+		return fmt.Sprintf("%s", domain)
+	}
+	return fmt.Sprintf("%s.%s", name, domain)
+}
+
+// subrouteURL returns the URL of the subroute given the scheme, traffic target name, and base domain. Curently
+// the subroute is represented as a subdomain of the base domain.
+func subrouteURL(scheme scheme, name, domain string) string {
+	return fmt.Sprintf("%s://%s", scheme, SubrouteDomain(name, domain))
+}
+
 // GetRevisionTrafficTargets returns a list of TrafficTarget flattened to the RevisionName, and having ConfigurationName cleared out.
-func (t *Config) GetRevisionTrafficTargets() []v1alpha1.TrafficTarget {
+func (t *Config) GetRevisionTrafficTargets(domain string) []v1alpha1.TrafficTarget {
 	results := make([]v1alpha1.TrafficTarget, len(t.revisionTargets))
 	for i, tt := range t.revisionTargets {
 		// We cannot `DeepCopy` here, since tt.TrafficTarget might contain both
 		// configuration and revision.
 		results[i] = v1alpha1.TrafficTarget{RevisionName: tt.RevisionName, Name: tt.Name, Percent: tt.Percent}
+		if tt.Name != "" && domain != "" {
+			// http is currently the only supported scheme
+			results[i].URL = subrouteURL(http, tt.Name, domain)
```

Comment:
I am afraid that using `http` here could be misleading as currently we have an [instruction](https://github.com/knative/docs/blob/master/docs/serving/using-an-ssl-cert.md) about configuring TLS.
If operators follow the instruction to configure TLS for some routes or subroutes, the `http` scheme could be confusing.

Reply:
Given that:

1. The current instructions are out of band of the Knative API
2. The current instructions still provides HTTP access through the ingress gateway
3. Auto TLS through the Knative API is coming out

I think I am okay with this. As soon as Knative has a signal that it can use to determine https is in-use this should be updated.

Thoughts?

Reply:
OK. SGTM